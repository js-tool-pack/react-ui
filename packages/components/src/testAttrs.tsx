import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
