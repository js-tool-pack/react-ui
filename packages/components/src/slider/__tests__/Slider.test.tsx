import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';
import { Slider } from '..';

describe('Slider', () => {
  testAttrs(Slider);

  test('basic', () => {
    const { container } = render(<Slider />);
    expect(container).toMatchSnapshot();
  });
});
