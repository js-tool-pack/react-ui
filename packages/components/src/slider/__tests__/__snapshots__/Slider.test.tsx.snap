// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider basic 1`] = `
<div>
  <div
    class="t-slider"
  >
    <div
      class="t-slider__rail"
    >
      <div
        class="t-slider__bar"
        style="left: 0%; width: 0%;"
      />
      <div
        class="t-slider-handlers"
      >
        <div
          class="t-slider-handlers__handler"
          draggable="false"
          style="left: 0%;"
          tabindex="1"
        />
      </div>
    </div>
  </div>
</div>
`;
